packages/gen-ai/frontend/src/app/services/llamaStackService.ts

```diff
@@ -73,709 +112,345 @@ const transformBackendResponse = (backendResponse: BackendResponseData): Simplif
+  ): ModArchRestDELETEWithoutData<T> =>
+  (queryParams: Record<string, unknown> = {}, opts: APIOptions = {}) =>
+    handleRestFailures(
+      restDELETE<T>(hostPath, path, {}, { ...baseQueryParams, ...queryParams }, opts),
+    ).then((response) => {
+      if (isModArchResponse<T>(response)) {
+        return response.data;
+      }
+      throw new Error('Invalid response format');
     });
-};
 
-export const deleteLSD = (project: string, lsdName: string): Promise<string> => {
-  const url = `${URL_PREFIX}/api/v1/lsd/delete?namespace=${project}`;
-  return axiosInstance
-    .delete(url, {
-      data: { name: lsdName },
-    })
-    .then((response) => response.data)
-    .catch((error) => {
-      throw new Error(
-        error.response?.data?.error?.message || error.message || 'Failed to delete LSD',
-      );
-    });
+// Axios-backed CREATE for non-JSON payloads (e.g., multipart/form-data), aligned with mod-arch shape
+const buildApiUrl = (
+  hostPath: string,
+  path: string,
+  queryParams: Record<string, unknown> = {},
+): string => {
+  const base = hostPath && hostPath.length > 0 ? hostPath : URL_PREFIX;
+  const qs = new URLSearchParams();
+  Object.entries(queryParams).forEach(([key, value]) => {
+    if (value !== undefined && value !== null) {
+      if (Array.isArray(value)) {
+        value.forEach((v) => qs.append(key, String(v)));
+      } else if (typeof value === 'object') {
+        qs.append(key, JSON.stringify(value));
+      } else if (
+        typeof value === 'string' ||
+        typeof value === 'number' ||
+        typeof value === 'boolean'
+      ) {
+        qs.append(key, String(value));
+      }
+    }
+  });
+  const queryString = qs.toString();
+  return `${base}${path}${queryString ? `?${queryString}` : ''}`;
 };
 
-/**
- * Fetches MCP servers from the cluster ConfigMap for a specific namespace
- * @param namespace - The namespace (project) to fetch MCP servers from
- * @returns Promise<MCPServersResponse> - MCP servers data with ConfigMap metadata
- * @throws Error - When the API request fails, ConfigMap doesn't exist, or insufficient permissions
- */
-export const getMCPServers = (namespace: string): Promise<MCPServersResponse> => {
-  if (!namespace || namespace.trim() === '') {
-    throw new Error('Namespace parameter is required');
-  }
+const axiosCREATE =
+  <T, D>(path: string) =>
+  (hostPath: string, baseQueryParams: Record<string, unknown> = {}): ModArchRestCREATE<T, D> =>
+  (data: D, opts: APIOptions = {}) =>
+    axiosInstance
+      .post(buildApiUrl(hostPath, path, baseQueryParams), data, opts)
+      .then((response) => {
+        const body = response.data;
+        if (isModArchResponse<T>(body)) {
+          return body.data;
+        }
+        throw new Error('Invalid response format');
+      })
+      .catch((error) => {
+        const message = error.response?.data?.error?.message || error.message || 'Request failed';
+        throw new Error(message);
+      });
```

Comment:
I cannot fully get rid of the Axios here because the shared package doesn't support `Content-Type` other than `application/json`, we might need to update the shared package.
cc @rohitkrai03